Gemfile

```diff
@@ -112,6 +112,7 @@ group :test do
   gem 'rspec-core'
   gem 'rspec-rerun'
   gem 'i18n-spec'
+  gem 'rails-controller-testing'
```

Comment:
Honestly, considering what is being said in the linked [deprecation message](https://github.com/rails/rails/issues/18950#issue-57746192), I'm not sure, if it's a good idea to use this gem at all:
> Testing what instance variables are set by your controller is a bad idea. 
> [...]The same goes for assert_template.

But I don't feel very strongly about it, so if you really want to use it and noone else has objections, I'm fine with it!

Reply:
I think you are right, we should try to conform with the rails conventions.
I'll rework the controller tests and remove tests that step out of the boundaries and try to add the parts as integration tests.